packages/cursorless-vscode/src/keyboard/grammar/grammar.ne

```diff
@@ -1,43 +1,35 @@
 @preprocessor typescript
 @{%
-import { capture, command, UNUSED as _ } from "../grammarHelpers"
+import { capture, command, UNUSED as _, argPositions } from "../grammarHelpers"
 import { keyboardLexer } from "../keyboardLexer";
+
+const { $0, $1, $2 } = argPositions;
 %}
 @lexer keyboardLexer
 
 # ===================== Top-level commands ===================
+
+# --------------------------- Marks --------------------------
 # "air"
-main -> decoratedMark {% command("targetDecoratedMarkReplace", ["decoratedMark"]) %}
+main -> decoratedMark {%
+  command("targetDecoratedMark", { decoratedMark: $0, mode: "replace" })
+%}
 
 # "past air"
 main -> %makeRange decoratedMark {%
-  command("targetDecoratedMarkExtend", [_, "decoratedMark"])
+  command("targetDecoratedMark", { decoratedMark: $1, mode: "extend" })
 %}
 
 # "and air"
 main -> %makeList decoratedMark {%
-  command("targetDecoratedMarkAppend", [_, "decoratedMark"])
+  command("targetDecoratedMark", { decoratedMark: $1, mode: "append" })
 %}
 
-# "funk"
-main -> scopeType {% command("modifyTargetContainingScope", ["scopeType"]) %}
-
-# "every funk"
-main -> %every scopeType {% command("targetEveryScopeType", [_, "scopeType"]) %}
+# --------------------------- Modifier --------------------------
 
-# "[third] next [two] funks"
-# "[third] previous [two] funks"
-main -> offset:? %nextPrev number:? scopeType {%
-  command(
-    "targetRelativeExclusiveScope",
-    ["offset", _, "length", "scopeType"],
-  )
-%}
+main -> modifier {% command("modifyTarget", { modifier: $0 }) %}
```

Comment:
I found myself wanting a railroad diagram again as I reviewed this.

Reply:
Every PR comes with one https://deploy-preview-2169--cursorless.netlify.app/keyboard-modal-railroad

Reply:
Nice! That’s helpful, thanks.